packages/Search/test/tstLinearBVH.cpp

```diff
@@ -41,134 +41,289 @@ void check_results( DataTransferKit::BVH<DeviceType> &bvh,
 {
-    Kokkos::View<DataTransferKit::Box *, DeviceType> boxes( "boxes", 2 );
-    auto boxes_host = Kokkos::create_mirror_view( boxes );
-    boxes_host( 0 ) = DataTransferKit::Box( {2., 2., 4., 4., 6., 6.} );
-    boxes_host( 1 ) = DataTransferKit::Box( {1., 1., 3., 3., 5., 5.} );
-    Kokkos::deep_copy( boxes, boxes_host );
-    DataTransferKit::BVH<DeviceType> bvh( boxes );
-    auto bounds = bvh.bounds();
-    TEST_EQUALITY( bounds[0], 1.0 );
-    TEST_EQUALITY( bounds[1], 2.0 );
-    TEST_EQUALITY( bounds[2], 3.0 );
-    TEST_EQUALITY( bounds[3], 4.0 );
-    TEST_EQUALITY( bounds[4], 5.0 );
-    TEST_EQUALITY( bounds[5], 6.0 );
+    // auto bvh = make_bvh<DeviceType>( {
+    //    {{2., 2., 4., 4., 6., 6.}},
+    //    {{1., 1., 3., 3., 5., 5.}},
+    //} );
+    // test_box_equality( bvh.bounds(), {{1., 2., 3., 4., 5., 6.}}, success, out
+    // );
+
+    // tree is empty, it has no leaves.
+    auto empty_bvh = make_bvh<DeviceType>( {} );
+
+    TEST_ASSERT( empty_bvh.empty() );
+    TEST_EQUALITY( empty_bvh.size(), 0 );
+    test_box_equality( empty_bvh.bounds(), {}, success, out );
+
+    // empty bvh no query
+    check_results( empty_bvh, make_overlap_queries<DeviceType>( {} ), {}, {0},
+                   success, out );
+
+    check_results( empty_bvh, make_within_queries<DeviceType>( {} ), {}, {0},
+                   success, out );
+
+    check_results( empty_bvh, make_nearest_queries<DeviceType>( {} ), {}, {0},
+                   success, out );
+
+    // empty bvh couple queries
+    check_results(
+        empty_bvh,
+        make_overlap_queries<DeviceType>( {
+            {}, // did not bother giving a valid box here but that's fine.
+            {},
+        } ),
+        {}, {0, 0, 0}, success, out );
+
+    check_results( empty_bvh,
+                   make_within_queries<DeviceType>( {
+                       {{{0., 0., 0.}}, 1.},
+                       {{{1., 1., 1.}}, 2.},
+                   } ),
+                   {}, {0, 0, 0}, success, out );
+
+    // NOTE: expected output for nearest queries when asking for more neighbors
+    // than they are leaves in the tree is not ideal but that is what the
+    // current implementation does to avoid to passes in the search (count
+    // first, then allocate and save results in 2nd pass) -1 denotes invalid
+    // index and infty invalid distance.
+    check_results( empty_bvh,
+                   make_nearest_queries<DeviceType>( {
+                       {{{0., 0., 0.}}, 1},
+                       {{{1., 1., 1.}}, 2},
+                   } ),
+                   {-1, -1, -1}, {0, 1, 3}, success, out );
+
+    double const infty = std::numeric_limits<double>::max();
+    check_results( empty_bvh,
+                   make_nearest_queries<DeviceType>( {
+                       {{{0., 0., 0.}}, 1},
+                       {{{1., 1., 1.}}, 2},
+                   } ),
+                   {-1, -1, -1}, {0, 1, 3}, {infty, infty, infty}, success,
```

Comment:
Should the distance be infty or negative? It seems easier to check that it is negative.

Reply:
My personal preference would be to change the behavior and return `{}, {0}, {}`. I want us to have that discussion and I will change the behavior in a separate PR.